pandas/_libs/src/ujson/python/date_conversions.c

```diff
@@ -79,7 +79,9 @@ char *PyDateTimeToIso(PyObject *obj, NPY_DATETIMEUNIT base,
     npy_datetimestruct dts;
     int ret;
 
-    ret = convert_pydatetime_to_datetimestruct(obj, &dts);
+    PyDateTime_IMPORT;
```

Comment:
Looking at the cpython source the datetime library seems to be implemented as a capsule. There's some documentaiton on that here:

https://docs.python.org/3/extending/extending.html#providing-a-c-api-for-an-extension-module

I really don't want to keep scattering these imports throughout the code base because it goes against the CPython documentation - can you check if the capsule approach would work?